flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -2562,19 +2505,26 @@ private StreamGraphGenerator getStreamGraphGenerator(List<Transformation<?>> tra
                     "No operators defined in streaming topology. Cannot execute.");
         }
 
+        // Synchronize the cached file to config option PipelineOptions.CACHED_FILES
+        if (!getCachedFiles().isEmpty()) {
```

Comment:
Because the cacheFile object is exposed to users and users can add items externally, so we could not migrate it to configuration. I'll add some comment for this change.